src/server_manager/web_app/digitalocean_app/create_server_app.ts

```diff
@@ -0,0 +1,75 @@
+/*
+  Copyright 2018 The Outline Authors
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+import {customElement, html, LitElement, property} from 'lit-element';
+
+import {ManagedServerRepository} from '../../model/server';
+import {OutlineNotificationManager} from '../ui_components/outline-notification-manager';
+import {Location, OutlineRegionPicker} from '../ui_components/outline-region-picker-step';
+
+// DigitalOcean mapping of regions to flags
+const FLAG_IMAGE_DIR = 'images/flags';
+const DIGITALOCEAN_FLAG_MAPPING: {[cityId: string]: string} = {
+  ams: `${FLAG_IMAGE_DIR}/netherlands.png`,
+  sgp: `${FLAG_IMAGE_DIR}/singapore.png`,
+  blr: `${FLAG_IMAGE_DIR}/india.png`,
+  fra: `${FLAG_IMAGE_DIR}/germany.png`,
+  lon: `${FLAG_IMAGE_DIR}/uk.png`,
+  sfo: `${FLAG_IMAGE_DIR}/us.png`,
+  tor: `${FLAG_IMAGE_DIR}/canada.png`,
+  nyc: `${FLAG_IMAGE_DIR}/us.png`,
+};
+
+@customElement('digital-ocean-create-server')
+export class DigitalOceanCreateServer extends LitElement {
+  @property({type: Function}) localize: Function;
+  @property({type: Object}) notificationManager: OutlineNotificationManager = null;
+
+  render() {
+    return html`
+        <outline-region-picker-step id="regionPicker" 
+            .localize=${this.localize}></outline-region-picker-step>
+    `;
+  }
+
+  // The region picker initially shows all options as disabled. Options are enabled by this code,
+  // after checking which regions are available.
+  async show(
+      digitalOceanRepository: ManagedServerRepository,
+      retryFn: <T>(fn: () => Promise<T>) => Promise<T>) {
```

Comment:
The idea we discussed before was to not do a retry here, and instead emit an auth error that the app.ts would handle. That would make the flow cleaner and easier to understand. Also more consistent with the rest of the app.

Did you figure out if you can detect auth errors?